src/components/live-tag/live-tag.tsx

```diff
@@ -7,49 +7,111 @@ export enum LiveTagStates {
   Preview = 'Preview',
 }
 
-export interface LiveTagProps {
-  state: LiveTagStates;
+interface LiveTagProps {
+  liveTagState: LiveTagStates;
+}
+
+interface LiveTagState {
+  liveTagState: LiveTagStates;
   isOnLiveEdge: boolean;
-  onClick: () => void;
 }
 
-export class LiveTag extends Component<LiveTagProps> {
-  shouldComponentUpdate(nextProps: LiveTagProps) {
+interface Context {
+  player: any;
+}
+
+export class LiveTag extends Component<LiveTagProps, LiveTagState> {
+  constructor(props: LiveTagProps, context: Context) {
+    super();
+    this.state = {
+      liveTagState: props.liveTagState,
+      isOnLiveEdge: context.player.isOnLiveEdge(),
+    };
+  }
+
+  shouldComponentUpdate(nextProps: LiveTagProps, nextState: LiveTagState) {
     return (
-      this.props.state !== nextProps.state ||
-      this.props.isOnLiveEdge !== nextProps.isOnLiveEdge
+      nextState.liveTagState !== this.state.liveTagState ||
+      nextState.isOnLiveEdge !== this.state.isOnLiveEdge
+    );
+  }
+
+  componentWillMount() {
+    this.context.player.addEventListener(
+      this.context.player.Event.SEEKED,
+      this._updateIsOnLiveEdge
+    );
+    this.context.player.addEventListener(
+      this.context.player.Event.PLAYING,
+      this._updateIsOnLiveEdge
+    );
+    this.context.player.addEventListener(
+      this.context.player.Event.PAUSE,
+      this._updateIsOnLiveEdge
     );
   }
+  componentWillUnmount() {
+    this.context.player.removeEventListener(
+      this.context.player.Event.SEEKED,
+      this._updateIsOnLiveEdge
+    );
+    this.context.player.removeEventListener(
+      this.context.player.Event.PLAYING,
+      this._updateIsOnLiveEdge
+    );
+    this.context.player.removeEventListener(
+      this.context.player.Event.PAUSE,
+      this._updateIsOnLiveEdge
+    );
+  }
+
+  public updateLiveTagState = (state: LiveTagStates) => {
+    this.setState({
+      liveTagState: state,
+    });
+  };
+
+  private _updateIsOnLiveEdge = () => {
+    this.setState({
+      isOnLiveEdge: this.context.player.paused
+        ? false
+        : this.context.player.isOnLiveEdge(),
```

Comment:
`player.isOnLiveEdge()` already should return `false` for `paused`. is not?

Reply:
No, the player.isOnLiveEdge() returns true for paused. I achieved result only with setTimeout
`
setTimeout(() => {
      this.setState({
        isOnLiveEdge: this.context.player.isOnLiveEdge(),
      });
    });
`

Reply:
totally agree it's better than timeout, but seems it's bug in the core impl. 
can you investigate why [this line](https://github.com/kaltura/playkit-js/blob/v0.67.0/src/player.js#L2052) doesn't work?

Reply:
listener for `PAUSE` event in kaltura-live plugin triggers earlier than listener in player.js
https://github.com/kaltura/playkit-js/blob/master/src/player.js#L1731
that's why `_isOnLiveEdge ` property is true